### PR TITLE
Maintenance

### PR DESCRIPTION
**Pull request by steelbrain**
- [x] Add support for `.bash_profile`
- [x] Add flow
- [x] Add eslint
- [x] Run both flow and eslint on CI

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "extends": "steelbrain",
+  "rules": {
+    "no-param-reassign": 0,
```
Comment (Arcanemagus) [inline .eslintrc.json]: `0` -> `"off"`

File: .flowconfig [hunk 2]
```diff
@@ -0,0 +1,12 @@
+[ignore]
+lib
```
Comment (Arcanemagus) [inline .flowconfig]: `<PROJECT_ROOT>/lib/.*`?

File: circle.yml [hunk 3]
```diff
@@ -1,17 +1,21 @@
...
     - npm run compile
 
 test:
   override:
-    - apm test
+    - npm test
+
+machine:
+  node:
+    version: 6.3.0
```
Comment (Arcanemagus) [inline circle.yml]: If you just do `6` it will install the latest v6 version

Comment (steelbrain) [reply circle.yml]: Gotcha

File: circle.yml [hunk 4]
```diff
@@ -1,17 +1,21 @@
...
     - sudo apt-get update
     - sudo apt-get -f install
-    - nvm use "v5.1.0"
-    - npm install --development
+    - node --version
+    - npm --version
+    - atom --version
+    - npm prune
+    - npm install
+    - apm rebuild
```
Comment (Arcanemagus) [inline circle.yml:12]: What's the point of this line?

File: package.json [hunk 5]
```diff
@@ -4,8 +4,10 @@
   "description": "A reliable way to get env",
   "main": "lib/index.js",
   "scripts": {
-    "test": "apm test",
-    "compile": "ucompiler go"
+    "test": "(apm test) && (flow check) && (eslint . )",
```
Comment (Arcanemagus) [inline package.json:7]: Maybe move the lint tasks to a different task, and then call that here if you still want that in `test`?

File: circle.yml (hunk 4 above)
Comment (steelbrain) [reply circle.yml:12]: It rebuilds any native node_modules with the correct version of NaN and stuff. It's required to have flat dependency structure of npm@3 (to make eslint work) while still making sure the native modules are compiled with apm

File: spec/.eslintrc.json [hunk 6]
```diff
@@ -0,0 +1,5 @@
+{
+  "env": {
+    "jasmine": true
```
Comment (Arcanemagus) [inline spec/.eslintrc.json:3]: Add `atomtest` for the `waitsForPromise()`

File: package.json (hunk 5 above)
Comment (steelbrain) [reply package.json:7]: Oh boy, I really really like this as one task 😛 
Besides when you are developing with Atom you do linting in it anyway, and when you are in console, you wanna test everything with just one command

File: spec/.eslintrc.json (hunk 6 above)
Comment (steelbrain) [reply spec/.eslintrc.json:3]: No need 'cause we are not using `waitsForPromise` anymore, we're using `async` `it`s now

File: circle.yml (hunk 4 above)
Comment (Arcanemagus) [reply circle.yml:12]: Ah, I just do `apm install --production` then `npm install` afterwards to install the devDependencies with `npm`.

File: package.json (hunk 5 above)
Comment (Arcanemagus) [reply package.json:7]: Was a suggestion, `npm run lint` is somewhat of a standard, but it isn't necessary 😛.

File: circle.yml (hunk 4 above)
Comment (steelbrain) [reply circle.yml:12]: I've actually tried that, but `fsevents` like dependencies are dev ones and npm compiles them with its lib versions

File: package.json (hunk 5 above)
Comment (steelbrain) [reply package.json:7]: Thank You ❤️

**Comment by Arcanemagus** [comment]
LGMT once CI passes 😛.

**Pull request merged**